File: tools/sumolib/color.py
# Eclipse SUMO, Simulation of Urban MObility; see https://eclipse.org/sumo
# Copyright (C) 2012-2023 German Aerospace Center (DLR) and others.
# This program and the accompanying materials are made available under the
# terms of the Eclipse Public License 2.0 which is available at
# https://www.eclipse.org/legal/epl-2.0/
# This Source Code may also be made available under the following Secondary
# Licenses when the conditions for such availability set forth in the Eclipse
# Public License 2.0 are satisfied: GNU General Public License, version 2
# or later which is available at
# https://www.gnu.org/licenses/old-licenses/gpl-2.0-standalone.html
# SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-or-later

# @file    color.py
# @author  Daniel Krajzewicz
# @author  Michael Behrisch
# @date    2012-12-04

from __future__ import absolute_import


class RGBAColor:

    def __init__(self, r, g, b, a=None):
        self.r = r
        self.g = g
        self.b = b
        self.a = a

    def toXML(self):
        if self.a is not None:
            return "%s,%s,%s,%s" % (self.r, self.g, self.b, self.a)
        else:
            return "%s,%s,%s" % (self.r, self.g, self.b)


def decodeXML(c):
    colorSwitch = {
            "red": RGBAColor(255, 0, 0, 255),
            "green": RGBAColor(0, 255, 0, 255),
            "blue": RGBAColor(0, 0, 255, 255),
            "yellow": RGBAColor(255, 255, 0, 255),
            "cyan": RGBAColor(0, 255, 255, 255),
            "magenta": RGBAColor(255, 0, 255, 255),
            "orange": RGBAColor(255, 128, 0, 255),
            "white": RGBAColor(255, 255, 255, 255),
            "black": RGBAColor(0, 0, 0, 255),
            "grey": RGBAColor(128, 128, 128, 255),
            "gray": RGBAColor(128, 128, 128, 255),
            "invisible": RGBAColor(0, 0, 0, 0)
            }
    knownColor = colorSwitch.get(c)
    if knownColor is None:
        return RGBAColor(*[float(x) for x in c.split(",")])
    return knownColor

File: tools/contributed/lisum-core/src/main/java/de/dlr/ts/commons/utils/print/Color.java
/****************************************************************************/
// Eclipse SUMO, Simulation of Urban MObility; see https://eclipse.org/sumo
// Copyright (C) 2016-2019 German Aerospace Center (DLR) and others.
// This program and the accompanying materials
// are made available under the terms of the Eclipse Public License v2.0
// which accompanies this distribution, and is available at
// http://www.eclipse.org/legal/epl-v20.html
// SPDX-License-Identifier: EPL-2.0
/****************************************************************************/
/// @file    Color.java
/// @author  Maximiliano Bottazzi
/// @date    2016
/// @version $Id$
///
//
/****************************************************************************/
package de.dlr.ts.commons.utils.print;

/**
 *
 * @author <a href="mailto:maximiliano.bottazzi@dlr.de">Maximiliano Bottazzi</a>
 */
public enum Color {
    NONE("", ""),
    BLANK("30", "40"),
    RED("31", "41"),
    GREEN("32", "42"),
    YELLOW("33", "43"),
    BLUE("34", "44"),
    MAGENTA("35", "45"),
    CYAN("36", "46"),
    WHITE("37", "47"),
    ORANGE("38;5;214", "48;5;208"),
    REDYELLOW("38;5;214", "48;5;208"),
    OFF("37", "47");

    private final String fore;
    private final String back;

    private Color(String fore, String back) {
        this.fore = fore;
        this.back = back;
    }

    public String getForegroundCode() {
        return fore;
    }

    public String getBackgroundCode() {
        return back;
    }
}
